service/controller/v3/resource/instance/create.go

```diff
@@ -9,116 +9,479 @@ import (
 	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
 	"github.com/giantswarm/microerror"
 
+	"github.com/giantswarm/azure-operator/service/controller/v3/controllercontext"
 	"github.com/giantswarm/azure-operator/service/controller/v3/key"
 )
 
+const (
+	masterVersionsKey  = "masterVersionBundleVersions"
+	versionsKey        = "versionBundleVersions"
+	vmssDeploymentName = "cluster-vmss-template"
+	workerVersionsKey  = "workerVersionBundleVersions"
+)
+
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
```

Comment:
Here happens all the magic now. The diff is weird. Maybe check the whole file. I think here is nothing we can just split up. This is one thing that works that way.